### PR TITLE
CSHARP-4528: Extend Search.Equals to support number and date

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: src/MongoDB.Driver/Search/OperatorSearchDefinitions.cs [hunk 1]
```diff
@@ -91,18 +91,36 @@ private protected override BsonDocument RenderArguments(IBsonSerializer<TDocumen
...
+                sbyte v => (BsonInt32)v,
+                byte v => (BsonInt32)v,
+                short v => (BsonInt32)v,
+                ushort v => (BsonInt32)v,
+                int v => (BsonInt32)v,
+                uint v => (BsonInt64)v,
+                long v => (BsonInt64)v,
+                float v => (BsonDouble)v,
+                double v => (BsonDouble)v,
+                DateTime v => (BsonDateTime)v,
```
Comment (JamesKovacs) [inline src/MongoDB.Driver/Search/OperatorSearchDefinitions.cs:120]: Should we support `DateTimeOffset` too?

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Spelling correction and minor question. Overall looks good.

File: src/MongoDB.Driver/Search/OperatorSearchDefinitions.cs (hunk 1 above)
Comment (BorisDog) [reply src/MongoDB.Driver/Search/OperatorSearchDefinitions.cs:120]: We can. We don't have a direct implicit conversion to `BsonDateTime`, so not sure how best to account for offset in this case.
Maybe we can consider support for `DateTimeOffset` both for equals and `Range` in a separate ticket?

File: tests/MongoDB.Driver.Tests/Search/SearchDefinitionBuilderTests.cs [hunk 2]
```diff
@@ -176,41 +177,65 @@ public void Compound_typed()
...
+                $"{{ equals: {{ path: 'x', value: {valueRendered}, score: {{ constant: {{ value: 1 }} }} }} }}");
+
+            AssertRendered(
+                subjectTyped.Equals(fieldExpression, value),
+                $"{{ equals: {{ path: '{fieldRendered}', value: {valueRendered} }} }}");
         }
 
-        [Fact]
-        public void Equals_typed()
+        public static object[][] EqualsSupporteddTypesTestData => new[]
```
[the remark this answers is not in the document]
Comment (BorisDog) [reply tests/MongoDB.Driver.Tests/Search/SearchDefinitionBuilderTests.cs]: Thanks, done.

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request merged**